DecipherNow/gm-fabric-dashboard
# Feat: Reusable Table Toolbar Component

Resolves #1067 
Resolves #1068 
Resolves #1069 
Resolves #1042 
Resolves #1021 
Resolves #992 

## Proposed Changes

  - There is now one TableToolbar that renders the appropriate subcomponents (search, display toggles, dropdowns) based on the props it receives. It is also possible to bypass these standard subcomponents altogether by using the toolbarChildren props to pass any custom element or node to the corresponding column.
- Deleted old toolbars and tests
- Added new unit test
- Added story (_Note: I still can't get styles to load for GMSelect.js and Button.js in storybook_)

## How to Test
- run `npm start` and verify the toolbars render and function as expected in the following areas:
    - Fabric Main View
    - Fabric Status Views (Stable, Warning, Down)
    - Service View
    - Go -> Functions & Routes
    - JVM -> Routes
 _Note: ThreadsTable was not touched, as it needs to be further refactored (see #610)_
- Type something into any toolbar search input and hit enter. Verify that the page does not hard refresh.
- run `npm test` and verify TableToolbar tests pass
- run `npm run storybook` and verify TableToolbar story renders
